core/amber/src/main/python/core/models/RTupleExecutor.py

```diff
@@ -0,0 +1,193 @@
+import pickle
+import datetime
+import pyarrow as pa
+import rpy2
+import rpy2.rinterface as rinterface
+import rpy2.robjects as robjects
+from rpy2_arrow.arrow import converter as arrow_converter
+from rpy2.robjects import default_converter
+from rpy2.robjects.conversion import localconverter as local_converter
+from typing import Iterator, Optional, Union
+from core.models import Tuple, TupleLike, TableLike
+from core.models.operator import SourceOperator, TupleOperatorV2
+import warnings
+
+warnings.filterwarnings(action="ignore", category=UserWarning, module=r"rpy2*")
+
+
+def _convert_r_to_py(value: rpy2.robjects):
+    """
+    :param value: A value that is from one of rpy2's many types (from rpy2.robjects)
+    :return: A Python representation of the value, if convertable.
+        If not, it returns the value itself
+    """
+    if isinstance(value, robjects.vectors.BoolVector):
+        return bool(value[0])
+    if isinstance(value, robjects.vectors.IntVector):
+        return int(value[0])
+    if isinstance(value, robjects.vectors.FloatVector):
+        if isinstance(value, robjects.vectors.POSIXct):
+            return next(value.iter_localized_datetime())
+        else:
+            return float(value[0])
+    if isinstance(value, robjects.vectors.StrVector):
+        return str(value[0])
+    return value
+
+
+class RTupleExecutor(TupleOperatorV2):
+    """
+    An operator that can execute R code on R Lists (R's representation of a Tuple)
+    """
+
+    is_source = False
+
+    _combine_binary_and_non_binary_lists = robjects.r(
+        """
+        function(non_binary_list, binary_list) {
+            non_binary_list <- as.list(non_binary_list$as_vector())
+            return (c(non_binary_list, binary_list))
+        }
+        """
+    )
```

Comment:
can we preserve the order of the fields?

Reply:
Yes, I forgot to talk to you about this. Please see the comment further below around line 136.

Reply:
Update: this has been moved to RUDFUtils.py in lines 47-55